tests/workflows/test_rolling_release_workflow.py

```
@@ -121,6 +122,56 @@ def test_install_components_uses_only_matrix_target_rustc_dev() -> None:
+    jobs = parsed.get("jobs")
+    assert isinstance(jobs, dict), "rolling-release workflow must declare jobs"
+    publish_job = jobs.get("publish")
+    assert isinstance(publish_job, dict), (
+        "rolling-release workflow must declare publish job"
+    )
+    needs = publish_job.get("needs")
+    if isinstance(needs, str):
+        needs = [needs]
+    assert isinstance(needs, list) and "build-lints" in needs, (
+        "publish job must depend on build-lints"
+    )
```

Ruff (0.15.2)

[warning] Assertion should be broken down into multiple parts

(PT018)